game-core/src/main/java/games/strategy/engine/data/DefaultAttachment.java

```diff
@@ -106,11 +106,10 @@ protected String thisErrorMsg() {
    * @return null or the toString() of the field value.
    */
   public String getRawPropertyString(final String property) {
-    final Object obj = getPropertyMap().get(property);
-    if (obj == null) {
-      return null;
-    }
-    return obj.toString();
```

Comment:
I believe this implementation had a bug introduced in #3107.  The above code returns the string representation of the `MutableProperty` itself, whereas the legacy `PropertyUtil#getPropertyFieldObject()` would return the string representation of the property value.  I think that's corrected below.